src/AutoRest.CSharp/Common/Output/Models/Types/ObjectType.cs

```diff
@@ -15,16 +19,24 @@ internal abstract class ObjectType : TypeProvider
         private ObjectTypeConstructor? _serializationConstructor;
         private ObjectTypeConstructor? _initializationConstructor;
         private string? _description;
+        private IEnumerable<ModelMethodDefinition>? _methods;
+        private ObjectTypeDiscriminator? _discriminator;
 
         protected ObjectType(BuildContext context) : base(context)
         {
         }
+        protected ObjectType(string defaultNamespace, SourceInputModel? sourceInputModel):base(defaultNamespace, sourceInputModel)
+        {
+        }
 
+        public bool IsStruct => ExistingType?.IsValueType == true;
```

Comment:
I personally prefer this:
```suggestion
        public bool IsStruct => ExistingType?.IsValueType ?? false;
```
which is emphasizing that this defaults to false.